calico_containers/calico_ctl/config.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2015 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Usage:
+  calicoctl config felix <NAME> [<VALUE>|--remove] [--force]
+  calicoctl config bgp <NAME> [<VALUE>|--remove] [--force]
+  calicoctl config node bgp <NAME> [<VALUE>|--remove] [--force]
+
+Description:
+  Configure or show low-level component configuration for Felix and BGP.
+
+Options:
+ --remove  Remove the configuration entry.
+ --force   Force update of configuration entry even if key value is unknown,
+           or the value is not recognized as valid.
+
+Valid configuration:
+  Command         | <NAME>  | <VALUE>s
+------------------+---------+-----------------------------------------
+  config felix    | logging | none debug info warning error critical
+  config bgp      | logging | none debug info
+  config node bgp | logging | none debug info
+
+Warnings:
+  -  Changing the global BGP logging levels using the `calicoctl config bgp`
+     command may cause all BGP sessions to bounce potentially resulting in a
+     transient loss of service.  If you need to change the logging level for
+     BGP, it is recommended to change the levels on a per-node basis using
+     the `calicoctl config node bgp` command.
+"""
+import re
+import sys
+
+from etcd import EtcdKeyNotFound
+from pycalico.datastore import handle_errors
+from pycalico.datastore import CONFIG_PATH, BGP_HOST_PATH, BGP_GLOBAL_PATH
+
+from connectors import client
+from utils import print_paragraph
+from utils import hostname
+
+# Dictionaries providing look up between the configuration name, and a tuple
+# of (internal name, value regex string)
+FELIX_CONFIG_DATA = {
+    "logging": ("LogSeverityScreen", "none|debug|info|warning|error|critical")
+}
+BGP_CONFIG_DATA = {
+    "logging": ("LogSeverityScreen", "none|debug|info")
```

Comment:
I'll change to loglevel.  I'll also change the nice name in all cases to be loglevel.